keras_cv/models/classification/image_classifier.py

```diff
@@ -0,0 +1,138 @@
+from keras_cv.models.backbones.backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.classification.image_classifier_presets import (
+    classifier_presets,
+)
+from keras_cv.models.task import Task
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ImageClassifier(Task):
+    """Image classifier with pooling and dense layer prediction head.
+
+    Args:
+        backbone: `keras.Model` instance, the backbone architecture of the
+            classifier called on the inputs.
+        num_classes: int, number of classes to predict.
+        pooling: str, type of pooling layer. Must be one of "avg", "max".
+        activation: A `str` or callable. The activation function to
+            use on the Dense layer. Set `classifier_activation=None` to return
+            the logits of the "top" layer.
+
+    Example:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained classifier (e.g., for imagenet categories)
+    model = keras_cv.models.ImageClassifier.from_preset(
+        "resnet50_v2_imagenet_classifier",
+        activation="softmax",
+    )
+    output = model(input_data)
+
+    # Pretrained backbone
+    backbone = keras_cv.models.ResNet50V2Backbone.from_preset(
+        "resnet50_v2_imagenet",
+    )
+    model = keras_cv.models.ImageClassifier(
+        backbone=backbone,
+        num_classes=4,
+        activation="softmax",
+    )
+    output = model(input_data)
+
+    # Randomly initialized backbone with a custom config
+    model = keras_cv.models.ImageClassifier(
+        backbone=keras_cv.models.ResNet50V2Backbone(),
+        num_classes=4,
+        activation="softmax",
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        backbone,
+        num_classes,
+        pooling="avg",
+        activation=None,
+        **kwargs,
+    ):
+        if pooling == "avg":
+            pooling_layer = layers.GlobalAveragePooling2D(name="avg_pool")
+        elif pooling == "max":
+            pooling_layer = layers.GlobalMaxPooling2D(name="max_pool")
+        else:
+            raise ValueError(
+                f'`pooling` must be one of "avg", "max". Received: {pooling}.'
+            )
+        inputs = backbone.input
+        x = backbone(inputs)
+        x = pooling_layer(x)
+        outputs = layers.Dense(
+            num_classes,
+            activation=activation,
+            name="predictions",
+        )(x)
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs=inputs,
+            outputs=outputs,
+            **kwargs,
+        )
+        # All references to `self` below this line
+        self.backbone = backbone
+        self.num_classes = num_classes
+        self.activation = activation
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
```

Comment:
I assume `backbone` is serialized by `Task.get_config()`? Might be worth a comment

Reply:
Yes. Every `Task` has a backbone so no special handling required. I can switch if this is too confusing however.

Reply:
Moved